pkg/kv/kvserver/store_create_replica.go

```diff
@@ -25,10 +25,25 @@ import (
 
 var errRetry = errors.New("retry: orphaned replica")
 
-// getOrCreateReplica returns a replica for the given RangeID, creating an
-// uninitialized replica if necessary. The caller must not hold the store's
-// lock. The returned replica has Replica.raftMu locked and it is the caller's
-// responsibility to unlock it.
+// getOrCreateReplica returns a replica with the given replicaID for the given
+// rangeID, or roachpb.RaftGroupDeletedError if this replicaID has been deleted.
+// If the replica is not in memory, but also is not deleted, it creates and
```

Comment:
This could be confusing for some readers. You say at the beginning of the comment that this method returns a replica (if it doesn't return an error). You don't mention raftMu.
Now here you say that if we're actually creating the replica, we lock raftMu and leave it locked. But the truth is, raftMu is locked whenever you get a replica back from this method.

We should also be precise about what it means to "create" a replica. First of all, if we're creating one, we're always creating an uninitialized one (since initialized replicas are always resident in memory). An uninitialized  pre-existing replica may not be resident in memory or it could be. So creating here really can mean either instantiating the mem-object only, or truly creating the replica. They look the same in code and we could also remove one case if we always had existing uninited replicas resident in memory (your comment in https://github.com/cockroachdb/cockroach/pull/94131#discussion_r1055362788) basically, but for now let's just document the way it is. Having all united replicas resident in memory is not a change I'm keen on since if some deployments have leaked 10000s of it over the years I don't want them to suddenly cause issues when they upgrade into a newer version.

Reply:
Done (I think). PTAL.